src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -1654,11 +1909,40 @@ public static void TryAddLifecycleHook<T>(this Microsoft.Extensions.DependencyIn
 
 namespace Aspire.Hosting.Publishing
 {
+    public sealed partial class AfterPublishEvent : Eventing.IDistributedApplicationEvent
+    {
+        public AfterPublishEvent(System.IServiceProvider services, ApplicationModel.DistributedApplicationModel model) { }
+
+        public ApplicationModel.DistributedApplicationModel Model { get { throw null; } }
+
+        public System.IServiceProvider Services { get { throw null; } }
+    }
+
+    public sealed partial class BeforePublishEvent : Eventing.IDistributedApplicationEvent
+    {
+        public BeforePublishEvent(System.IServiceProvider services, ApplicationModel.DistributedApplicationModel model) { }
+
+        public ApplicationModel.DistributedApplicationModel Model { get { throw null; } }
+
+        public System.IServiceProvider Services { get { throw null; } }
+    }
+
     public partial interface IDistributedApplicationPublisher
     {
         System.Threading.Tasks.Task PublishAsync(ApplicationModel.DistributedApplicationModel model, System.Threading.CancellationToken cancellationToken);
     }
 
+    public partial interface IPublishingActivityProgressReporter
+    {
+        System.Threading.Tasks.Task<PublishingActivity> CreateActivityAsync(string id, string initialStatusText, bool isPrimary, System.Threading.CancellationToken cancellationToken);
+        System.Threading.Tasks.Task UpdateActivityAsync(PublishingActivity publishingActivity, System.Threading.CancellationToken cancellationToken);
+    }
+
+    public partial interface IResourceContainerImageBuilder
```

Comment:
Experimental

Reply:
@mitchdenny

Reply:
It is: https://github.com/dotnet/aspire/blob/1007a6c69ab3e2e1a09e06bbf02d53a1208a8c2e/src/Aspire.Hosting/Publishing/ResourceContainerImageBuilder.cs#L19